### PR TITLE
fix: fix capabilities check for PodDisrutionBudget

### PR DESCRIPTION
**Pull request by sharkannon**
### Checklist

- [x] Changelog updated or skip changelog label added
- [x] Documentation updated
- [x] Template tests added for new features
- [x] Integration tests added or modified for major features

### CODE REVIEW

**Comment by sharkannon** [comment]
Example of what I mean based on the value of Capabilities.APIVersions

Helm Template:

```
{{- range $key := $.Capabilities.APIVersions }}
- {{ . }}
{{- end }} 
```

Output:

```
- v1
- admissionregistration.k8s.io/v1
- admissionregistration.k8s.io/v1alpha1
- admissionregistration.k8s.io/v1beta1
- internal.apiserver.k8s.io/v1alpha1
- apps/v1
- apps/v1beta1
- apps/v1beta2
- authentication.k8s.io/v1
- authentication.k8s.io/v1alpha1
- authentication.k8s.io/v1beta1
- authorization.k8s.io/v1
- authorization.k8s.io/v1beta1
- autoscaling/v1
- autoscaling/v2
- autoscaling/v2beta1
- autoscaling/v2beta2
- batch/v1
- batch/v1beta1
- certificates.k8s.io/v1
- certificates.k8s.io/v1beta1
- certificates.k8s.io/v1alpha1
- coordination.k8s.io/v1beta1
- coordination.k8s.io/v1
- discovery.k8s.io/v1
- discovery.k8s.io/v1beta1
- events.k8s.io/v1
- events.k8s.io/v1beta1
- extensions/v1beta1
- flowcontrol.apiserver.k8s.io/v1
- flowcontrol.apiserver.k8s.io/v1beta1
- flowcontrol.apiserver.k8s.io/v1beta2
- flowcontrol.apiserver.k8s.io/v1beta3
- networking.k8s.io/v1
- networking.k8s.io/v1alpha1
- networking.k8s.io/v1beta1
- node.k8s.io/v1
- node.k8s.io/v1alpha1
- node.k8s.io/v1beta1
- policy/v1
- policy/v1beta1
- rbac.authorization.k8s.io/v1
- rbac.authorization.k8s.io/v1beta1
- rbac.authorization.k8s.io/v1alpha1
- resource.k8s.io/v1alpha2
- scheduling.k8s.io/v1alpha1
- scheduling.k8s.io/v1beta1
- scheduling.k8s.io/v1
- storage.k8s.io/v1beta1
- storage.k8s.io/v1
- storage.k8s.io/v1alpha1
- apiextensions.k8s.io/v1beta1
- apiextensions.k8s.io/v1
```

Note, `policy/v1/PodDisruptionBudget` doesn't exist, so the original check would always show "false"

File: .changelog/3514.fixed.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Issue #3513 fixing Capabilities logic, as policy/v1/PodDisrutionBudget is not a part of .Capabilities.APIVersions however policy/v1 is
```
Comment (sumo-drosiek) [inline .changelog/3514.fixed.txt:1]: fix: fix capabilities check for PodDisrutionBudget

**Review by sumo-drosiek — APPROVED** [review]
LGTM. Please update changelog entry

**Comment by sharkannon** [comment]
@sumo-drosiek Just updated the changelog to match what you requested.

**Pull request merged**